go/libraries/doltcore/sqle/tables.go

```diff
@@ -228,6 +258,7 @@ var _ sql.UpdatableTable = (*WritableDoltTable)(nil)
 var _ sql.DeletableTable = (*WritableDoltTable)(nil)
 var _ sql.InsertableTable = (*WritableDoltTable)(nil)
 var _ sql.ReplaceableTable = (*WritableDoltTable)(nil)
+var _ sql.AutoIncrementTable = (*WritableDoltTable)(nil)
```

Comment:
Ideally you would only implement this conditionally for tables that have autoincrement, but you can't really do that.

The other alternative is that you use some standard error type provided by go-mysql-server to return an error in the case that a table doesn't have any auto_increment columns.